From 5765e58d901e6ac9e9b68f47390ef70e16f7dccc Mon Sep 17 00:00:00 2001
From: Pekka Vuorela <pekka.vuorela@jolla.com>
Date: Tue, 21 Sep 2021 18:26:24 +0300
Subject: [PATCH 7/8] Update nfo:fileName on content specific graphs too when
 moved

Moved files were updating nfo:fileName only in tracker:FileSystem.

Fixes: https://gitlab.gnome.org/GNOME/tracker-miners/-/issues/194
---
 src/miners/fs/tracker-miner-files.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/miners/fs/tracker-miner-files.c b/src/miners/fs/tracker-miner-files.c
index fb7b3152f..d1c1dc841 100644
--- a/src/miners/fs/tracker-miner-files.c
+++ b/src/miners/fs/tracker-miner-files.c
@@ -2423,14 +2423,16 @@ miner_files_move_file (TrackerMinerFS      *fs,
 	                        "} INSERT {"
 	                        "  GRAPH ?g {"
 	                        "    <%s> a nfo:FileDataObject ; "
+	                        "         nfo:fileName \"%s\" ; "
 	                        "         ?p ?o "
 	                        "  }"
 	                        "} WHERE {"
 	                        "  GRAPH ?g {"
 	                        "    <%s> ?p ?o "
 	                        "  }"
+	                        "  FILTER (?p != nfo:fileName) . "
 	                        "}",
-	                        source_uri, uri, source_uri);
+	                        source_uri, uri, display_name, source_uri);
 	g_free (container_clause);
 
 	if (recursive) {
-- 
2.31.1

